pkg/controller/install/webhook.go

```diff
@@ -162,6 +188,144 @@ func (i *StrategyDeploymentInstaller) createOrUpdateValidatingWebhook(ogNamespac
+
+		// check if CRD exists on cluster
+		crd, err := i.strategyClient.GetOpClient().ApiextensionsInterface().ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), desc.ConversionCRDs[index], metav1.GetOptions{})
+
+		if err != nil {
+			log.Info("CRD not found %s, error: %s", desc.ConversionCRDs[index], err.Error())
+		}
+		log.Info("Found conversionCRDs %s", desc.ConversionCRDs[index])
+
+		// check if the CSV owns this CRD
+		foundCRD := false
+		csv, err := i.strategyClient.GetOpLister().OperatorsV1alpha1().ClusterServiceVersionLister().ClusterServiceVersions(i.owner.GetNamespace()).Get(i.owner.GetName())
+		if err != nil {
+			log.Info("CSV not found, error: %s", err.Error())
+		}
+		ownedCRDs := csv.Spec.CustomResourceDefinitions.Owned
+		for _, ownedCRD := range ownedCRDs {
+			if ownedCRD.Name == crd.GetName() {
+				log.Info("CSV %q owns CRD %q", csv.GetName(), crd.GetName())
+				foundCRD = true
+			}
+		}
+		if !foundCRD {
+			log.Info("CSV %q does not own CRD %q", csv.GetName(), crd.GetName())
+			return nil
+		}
+
+		// crd.Spec.Conversion.Strategy specifies how custom resources are converted between versions.
+		// Allowed values are:
+		// 	- None: The converter only change the apiVersion and would not touch any other field in the custom resource.
+		// 	- Webhook: API Server will call to an external webhook to do the conversion. This requires crd.Spec.preserveUnknownFields to be false.
+		// References:
+		//  - https://docs.openshift.com/container-platform/4.5/rest_api/extension_apis/customresourcedefinition-apiextensions-k8s-io-v1.html
+		// 	- https://kubernetes.io/blog/2019/06/20/crd-structural-schema/#pruning-don-t-preserve-unknown-fields
+		// By default the strategy is none
+		// Reference:
+		// 	- https://v1-15.docs.kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#specify-multiple-versions
+		if crd.Spec.PreserveUnknownFields == true && crd.Spec.Conversion.Strategy == "Webhook" {
+			log.Info("crd.Spec.PreserveUnknownFields must be false to let API Server call webhook to do the conversion.")
+			return nil
+		}
+
+		// Conversion WebhookClientConfig need not be set when Strategy is None
+		// https://v1-15.docs.kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#specify-multiple-versions
+		// Conversion WebhookClientConfig needs to be set when Strategy is None
+		// https://v1-15.docs.kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#configure-customresourcedefinition-to-use-conversion-webhooks
+		if crd.Spec.Conversion.Strategy == "Webhook" {
+			log.Info("Updating CRD")
+
+			// use user defined CRD conversion Webhook Strategy
+			conversionWebhookStrategy := crd.Spec.Conversion.Strategy
+
+			// use user defined name for CRD conversion webhook
+			conversionWebhookName := crd.Spec.Conversion.Webhook.ClientConfig.Service.Name
```

Comment:
I believe that OLM generates the service for the webhook on the fly - I don't believe that this will work.